armi/physics/neutronics/fissionProductModel/fissionProductModel.py

```diff
@@ -85,64 +95,50 @@ def _fissionProductBlockType(self):
-            this is a flag to tell the method whether or not to try to apply
-            the global lumped fission product to the component and thereby
-            material -- this is only compatable with
-            LumpedFissionProductCompatableMaterial
-
-        Examples
-        --------
-        self.setAllBlockLFPs(blockType='fuel')
-        will apply the global lumped fission product or independent LFPs to only
-        fuel type blocks
-
+        Sets all the block lumped fission products attributes and adds fission products
+        to each block if `self._explicitFissionProducts` is set to True.
         See Also
         --------
-        armi.materials.lumpedFissionProductCompatableMaterial.LumpedFissionProductCompatableMaterial
         armi.reactor.components.Component.setLumpedFissionProducts
-        armi.physics.neutronics.fissionProductModel.fissionProductModel.FissionProductModel.setAllBlockLFPs
         """
-        blockType = blockType or self._fissionProductBlockType
-        for b in self.r.core.getBlocks(blockType, includeAll=True):
+        for b in self.r.core.getBlocks(self._fissionProductBlockType, includeAll=True):
             if self._useGlobalLFPs:
                 b.setLumpedFissionProducts(self.getGlobalLumpedFissionProducts())
             else:
-                independentLFPs = self.getGlobalLumpedFissionProducts().duplicate()
-                b.setLumpedFissionProducts(independentLFPs)
-
-        # use two loops to pass setting the material LFPs
-        if setMaterialsLFP:
-            for b in self.r.core.getBlocks(blockType, includeAll=True):
-                if self._useGlobalLFPs:
-                    b.setChildrenLumpedFissionProducts(
-                        self.getGlobalLumpedFissionProducts()
-                    )
+                lfps = self.getGlobalLumpedFissionProducts()
+                if lfps is None:
+                    b.setLumpedFissionProducts(None)
                 else:
                     independentLFPs = self.getGlobalLumpedFissionProducts().duplicate()
-                    b.setChildrenLumpedFissionProducts(independentLFPs)
+                    b.setLumpedFissionProducts(independentLFPs)
+
+            # Initialize the fission products explicitly on the block component
+            # that matches the `self._fissionProductBlockType` if it exists.
+            if self._explicitFissionProducts and not self._initialized:
+                targetComponent = b.getComponent(self._fissionProductBlockType)
+                if not targetComponent:
+                    continue
+                ndens = targetComponent.getNumberDensities()
+                updatedNDens = {}
+                for nuc in self.r.blueprints.allNuclidesInProblem:
+                    if nuc in ndens:
+                        continue
+                    updatedNDens[nuc] = 0.0
+                targetComponent.updateNumberDensities(updatedNDens)
+        self._initialized = True
```

Comment:
This was added and paired with the addition of the `_autoUpdateNuclideFlags` method in `blueprints`. Essentially, moving between lumped fission product modeling and `explicitFissionProducts` would require the user to add 100s of fission products into the `nuclideFlags` section of the blueprints. This allows the same `blueprints` input to be used between two different cases without requiring a significant effort and migration of existing files.

This seemed like the best place to add the extra fission product nuclides to the blocks/components since this is the `FissionProductModel` interface. I tried a couple different ways to add these in the `componentBlueprints` or elsewhere and this seemed like the best fit so that the rest of the reactor construction doesn't have this one-off code based on a setting. 

Another point here is that settings are typically not passed all through the blueprints objects so it  made the code really messy to start passing a settings object around or adding more keyword arguments.